lib/ansible/plugins/inventory/linode.py

```diff
@@ -125,18 +131,39 @@ def _filter_by_config(self, regions, types):
                 if instance.type.id in types
             ]
 
-    def _add_instances_to_groups(self):
+    def _add_instances_to_groups(self, use_public_ip):
         """Add instance names to their dynamic inventory groups."""
         for instance in self.instances:
-            self.inventory.add_host(instance.label, group=instance.group)
+            if use_public_ip:
+                host_identifier = self._get_instance_public_ip(instance)
+            else:
+                host_identifier = instance.label
+            self.inventory.add_host(host_identifier, group=instance.group)
+
+    def _get_instance_public_ip(self, instance):
+        """Returns the public IP address of the instance"""
+        if instance.ipv4 and len(instance.ipv4) > 0:
+            for ip in instance.ipv4:
+                if not ipaddress.ip_address(ip).is_private:
+                    return ip
+            return False
+        else:
+            raise IndexError(
```

Comment:
This should use `AnsibleError` like the other errors to pass back a nicely formatted message to the user.

Reply:
Wasn't sure about using that here, will change.